File: trunk/ShadowFrameworkLite20/src/shadow/system/data/objects/SFBinaryValue.java
package shadow.system.data.objects;

public abstract class SFBinaryValue {

	protected int value;

	@Override
	public abstract SFBinaryValue clone();

	protected abstract int getBitSize();

	public SFBinaryValue() {
		super();
		this.value = getBitSize();
	}

	public int getValue() {
		return value;
	}

	protected void setValue(int value) {
		this.value = value;
	}

}

File: trunk/SFDataApplicationTestC++/shadow/system/data/objects/SFBinaryValue.cpp

#include "SFBinaryValue.h"

namespace sf{
	SFBinaryValue::SFBinaryValue() {
        //Solve this
		this->value=0;
        //this->value = this->getBitSize();
	}

	int SFBinaryValue::getValue() {
		return value;
	}

	void SFBinaryValue::setValue(int value) {
		this->value = value;
	}
}
